WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -607,6 +648,47 @@ public void onDismissed(Snackbar snackbar, int event) {
         snackbar.show();
     }
 
+    /**
+     * Searches the PostStore for post with a matching remote ID. Used when searching since PostModel's returned
+     * from search actions don't have valid local IDs
+     */
+    private PostModel getPostFromRemoteId(long remoteId) {
+        List<PostModel> posts;
+        if (!mIsPage) {
+            posts = mPostStore.getPostsForSite(mSite);
+        } else {
+            posts = mPostStore.getPagesForSite(mSite);
+        }
+        for (PostModel post : posts) {
+            if (post.getRemotePostId() == remoteId) {
+                return post;
+            }
+        }
+        return null;
+    }
+
+    private void dispatchSearchAction() {
+        SearchPostsPayload payload = new SearchPostsPayload(mSite, mSearchTerm, mSearchOffset);
+        if (mIsPage) {
+            mDispatcher.dispatch(PostActionBuilder.newSearchPagesAction(payload));
+        } else {
+            mDispatcher.dispatch(PostActionBuilder.newSearchPostsAction(payload));
+        }
+    }
+
+    private final Runnable mSearchRunnable = new Runnable() {
+        @Override
+        public void run() {
+            if (TextUtils.isEmpty(mSearchTerm)) {
```

Comment:
Assuming that we centralize the `search` method, it might make sense to move most of this code there and only keep what's absolutely necessary to be in the runnable.